### ISSUE TITLE
Disable Nautilus menu Shaddow

### ISSUE THREAD

**Issue by zhou13** [yshui/compton#1026]
### Platform
Archlinux

### GPU, drivers, and screen setup
nvidia-525.89.02

### Environment
i3

### picom version

<details>
<summary>Diagnostics</summary>
**Version:** vgit-b700a

### Extensions:

* Shape: Yes
* XRandR: Yes
* Present: Present

### Misc:

* Use Overlay: No
  (Another compositor is already running)
* Config file used: /home/yichaozhou/.config/picom.conf

### Drivers (inaccurate):

NVIDIA

### Backend: glx

* Driver vendors:
 * GLX: NVIDIA Corporation
 * GL: NVIDIA Corporation
* GL renderer: NVIDIA GeForce GT 1030/PCIe/SSE2

### Backend: egl

* Driver vendors:
 * EGL: NVIDIA
 * GL: NVIDIA Corporation
* GL renderer: NVIDIA GeForce GT 1030/PCIe/SSE2

</details>

### Configuration:
<details>
    <summary>Configuration file</summary>

```
backend = "xrender";

vsync = true;
unredir-if-possible = true;

shadow = true;
shadow-radius = 7;
shadow-offset-x = -7;
shadow-offset-y = -7;

# Window type settings
wintypes:
{
  tooltip = { fade = true; shadow = true; focus = true; };
  dock = { shadow = false; }
  dnd = { shadow = false; }
  utility = { shadow = false; };
};

shadow-exclude = [
  "class_g = 'firefox' && class_i = 'Popup' && width < 128",
  "class_g = 'Navigator' && class_i = 'Popup' && width < 128",
  "class_g = 'i3-frame'",
  "class_g = 'Wine'",
  "class_g = 'zoom'",
  "class_g = 'slop'",
  "class_g = 'looking-glass-client'",
  "name = 'cpt_frame_window'",
];


# vim: set ft=config:
```
</details>

### Steps of reproduction
1. Open nautilus
2. Right click

### Expected behavior

Right menu looks good.

### Current Behavior

Right menu looks bad.

![2023-02-23|00:12:54](https://user-images.githubusercontent.com/576382/220853639-0c6075c9-a7eb-45c1-8b7e-f1db1ae61d45.png)

### Other details

I can fix this by adding 
```
  menu = { shadow = false; };
```
to `wintypes`.  However, it will also disable shadow on all menu in other apps.

**Comment by absolutelynothelix** [comment]
```
shadow-exclude = [
  ...
  "window_type = 'menu' && %a rule to match nautilus' windows%",
  ...
];
```

**Comment by zhou13** [comment]
Thanks! It works very well.  For reference, you can use:
```
shadow-exclude = [
  "window_type = 'menu' && class_g = 'org.gnome.Nautilus'",
];
```
It would be best if this can be done automatically without extra configuration.

**Comment by absolutelynothelix** [comment]
unfortunately, we can’t predict what a user wants. i agree that the `wintypes` option is a little bit confusing, but there is some work done on unified configuration for windows (all of these `-exclude`, `-include` options and the `wintypes` option are supposed to be replaced with a single option).

**Comment by zhou13** [comment]
The unification of configuration would be very helpful. What I  mean is that Gnome's compositor can handle this kind of thing without any user configuration and disable shadow correctly. Maybe there is a way to detect this kind of window automatically (e.g., there is a flag somewhere)?

**Comment by absolutelynothelix** [comment]
ah, i got you. try this rule: https://github.com/yshui/picom/blob/next/picom.sample.conf#L51

i’m not sure, but iiuc it should handle this kind of windows

**Comment by z1atk0** [comment]
> ah, i got you. try this rule: https://github.com/yshui/picom/blob/next/picom.sample.conf#L51
> i’m not sure, but iiuc it should handle this kind of windows

No, that doesn't work here. I used the provided `sample.conf` with only a few changes (mainly disabling/reducing transparency), but that strange extra frame in Nautilus right-click popup windows is still there. Have not tried @zhou13's `shadow-exclude` rule yet, but will do then when I give `picom` another go (I switched back to Metacity's internal compositor for the time being).

**Comment by zhou13** [comment]
@z1atk0 I have tried https://github.com/yshui/picom/blob/next/picom.sample.conf#L51 and it also does not work for me either.

**Comment by z1atk0** [comment]
Oh, and I'm using `backend = "glx";` instead of the default `backend = "xrender";` on an NVIDIA GeForce GTX 1660 Ti with the 525.89.02 binary drivers ... just in case that makes a difference. :man_shrugging:

**Comment by z1atk0** [comment]
Just switched back to `picom` after Metacity's internal compositor caused two crashes within three days again, taking my whole desktop down with it, while I was in the middle of doing some work. :angry: And yes, @zhou13's `shadow-exclude` rule works fine with Nautilus (although I've meanwhile switched to Nemo, as Nautilus every now and then decides to take aaaaaaaaaaaages to simply open a (local) folder (on an SSD)  since they switched to GTK4, but that's a different story :roll_eyes:), thanks a lot! :+1: :sunglasses: 

I then found out that both Firefox's and Thunderbird's right-click menus actually have the same problem (ie. an additonal frame around the popup menu), and fixed it like this:
```
shadow-exclude = [
[...]
  "window_type = 'popup_menu' && class_g = 'firefox-esr'",
  "window_type = 'popup_menu' && class_g = 'thunderbird'",
[...]
];
```
Maybe this helps someone who has the same problem save some time on research and/or trial & error. :slightly_smiling_face:
